receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,311 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+)
+
+type logsReceiver struct {
+	log         *zap.Logger
+	cfg         *Config
+	client      *internal.MongoDBAtlasClient
+	consumer    consumer.Logs
+	stopperChan chan struct{}
+	wg          sync.WaitGroup
+}
+
+type resourceInfo struct {
+	Org      string
+	Project  mongodbatlas.Project
+	Cluster  mongodbatlas.Cluster
+	Hostname string
+	LogName  string
+	start    time.Time
+	end      time.Time
+}
```

Comment:
This struct could be problematic. It's a bag of contextual info, parts of which are updated by many different functions at many different levels. It's very difficult to conclude that there are no adverse side effects to this.

I think there actually may be some problems with the way the `end` time is being tracked. If I'm not mistaken, you are only instantiating a single instance of this so the `end` time ends up being clobbered each time a project is collected. If there's any meaningful time between the first and last projects being collected, then you've created a gap during which you will never collect logs on the first project.

Some thoughts:
- `end` should be tracked per project. Each project has an `Org` and a project id as well. These could all be combined into a `ProjectContext` which gets passed around. 
- `start` is only set once, so just attach this to the receiver. Alternately, you could put it in `ProjectContext` so that it rides alongside `end`, even though all instances of it will be the same. 
- `LogName` is only attached to this struct for the sake of passing it to `mongodbEventToLogData`. It's passed around separately otherwise. Just remove it from this struct and keep passing it around as necessary.
- `Hostname` is also only here for the sake of passing it through one layer (`collectLogs`/`collectAuditLogs`). Then it gets pulled right back out and passed around on its own again. Just remove it from this struct and pass it around as necessary.
- `Cluster` is basically the same thing. Just pass it around as necessary.

Basically, delete `resourceInfo`, replace it with `ProjectContext`, and pass everything else around as necessary:
```go
type ProjectContext struct {
	Project  mongodbatlas.Project
	orgName  string
	projectID string
	start    time.Time
	end      time.Time
}
```